Polyfills/Canvas/Source/Context.cpp

```diff
@@ -0,0 +1,694 @@
+            {"powderblue", 0xb0e0e6},
+            {"purple", 0x800080},
+            {"red", 0xff0000},
+            {"rosybrown", 0xbc8f8f},
+            {"royalblue", 0x4169e1},
+            {"saddlebrown", 0x8b4513},
+            {"salmon", 0xfa8072},
+            {"sandybrown", 0xf4a460},
+            {"seagreen", 0x2e8b57},
+            {"seashell", 0xfff5ee},
+            {"sienna", 0xa0522d},
+            {"silver", 0xc0c0c0},
+            {"skyblue", 0x87ceeb},
+            {"slateblue", 0x6a5acd},
+            {"slategray", 0x708090},
+            {"slategrey", 0x708090},
+            {"snow", 0xfffafa},
+            {"springgreen", 0x00ff7f},
+            {"steelblue", 0x4682b4},
+            {"tan", 0xd2b48c},
+            {"teal", 0x008080},
+            {"thistle", 0xd8bfd8},
+            {"tomato", 0xff6347},
+            {"turquoise", 0x40e0d0},
+            {"violet", 0xee82ee},
+            {"wheat", 0xf5deb3},
+            {"white", 0xffffff},
+            {"whitesmoke", 0xf5f5f5},
+            {"yellow", 0xffff00},
+            {"yellowgreen", 0x9acd32}};
+        
+        if (str[0] == '#' && str.length() == 4)
+        {
+            unsigned int components[4];
+            int count = sscanf(str.c_str(), "#%1x%1x%1x", &components[0], &components[1], &components[2]);
+            for (int i = count; count < 4; count++)
+            {
+                components[i] += components[i] << 4;
+            }
+            for (int i = count; count < 4; count++)
+            {
+                components[i] = 255;
+            }
+            return nvgRGBA(components[0], components[1], components[2], components[3]);
+        }
+        else if (str[0] == '#' && str.length() == 7)
+        {
+            unsigned int components[4];
+            int count = sscanf(str.c_str(), "#%02x%02x%02x%02x", &components[0], &components[1], &components[2], &components[3]);
+            for (int i = count; count < 4; count++)
+            {
+                components[i] = 255;
+            }
+            return nvgRGBA(components[0], components[1], components[2], components[3]);
+        }
+        else
+        {
+            auto iter = webColors.find(str);
+            if (iter != webColors.end())
+            {
+                uint32_t color = iter->second;
+                return nvgRGBA((color>>16), (color>>8)&0xFF, (color&0xFF), 0xFF);
+            }
+            else
+            {
+                assert(0);
+            }
+        }
+        return nvgRGBA(255, 0, 255, 255);
+    }
+
+    void Context::FillRect(const Napi::CallbackInfo& info)
+    {
+        auto left = info[0].As<Napi::Number>().FloatValue();
+        auto top = info[1].As<Napi::Number>().FloatValue();
+        auto width = info[2].As<Napi::Number>().FloatValue();
+        auto height = info[3].As<Napi::Number>().FloatValue();
+
+        NVGpaint paint = nvgLinearGradient(m_nvg, 0, 5, 0, 10, nvgRGBA(0, 160, 192, 255), nvgRGBA(0, 160, 192, 255));
```

Comment:
Should put comments on functions that are not complete yet.

Reply:
Or a comment at the top saying that these functions are not all done and some are done but not fully tested.